tracer/src/Datadog.Trace/Configuration/Schema/DatabaseSchema.cs

```diff
@@ -0,0 +1,41 @@
+// <copyright file="DatabaseSchema.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.Collections.Generic;
+
+namespace Datadog.Trace.Configuration.Schema
+{
+    internal class DatabaseSchema
+    {
+        private readonly SchemaVersion _version;
+        private readonly string _defaultServiceName;
+        private readonly IDictionary<string, string>? _serviceNameMappings;
+
+        public DatabaseSchema(SchemaVersion version, string defaultServiceName, IDictionary<string, string>? serviceNameMappings)
+        {
+            _version = version;
+            _defaultServiceName = defaultServiceName;
+            _serviceNameMappings = serviceNameMappings;
+        }
+
+        public string GetOperationName(string databaseType) => $"{databaseType}.query";
+
+        public string GetServiceName(string databaseType)
+        {
+            if (_serviceNameMappings is not null && _serviceNameMappings.TryGetValue(databaseType, out var mappedServiceName))
+            {
+                return mappedServiceName;
+            }
+
+            return _version switch
+            {
+                SchemaVersion.V1 => _defaultServiceName,
+                _ => $"{_defaultServiceName}-{databaseType}",
```

Comment:
Would it make sense to use "V0/else" instead of "V1/else"? I think only V0 will ever use `{_defaultServiceName}-{databaseType}` and all new schema versions (in the foreseeable future) will use `_defaultServiceName`. This would avoid us having to modify this code every time a new version is added.

```suggestion
                SchemaVersion.V0 => $"{_defaultServiceName}-{databaseType}",
                _ => _defaultServiceName,
```

(I'm assuming that if a schema version is not set explicitly, we fallback to V0.)

Reply:
Good point! I agree it makes sense to do "V0/else" since I don't expect our calculations to change. I think I got into the "V1/else" thinking during the configuration validation (where the fallback for any bad input is V0) and carried that everyone else. I'll update the PR with a small tweak

Reply:
Done in ad18079373115a00fb04e1a57bf81f39d1b8f420